distributed/client.py

```diff
@@ -2220,19 +2229,15 @@ async def wait(k):
 
             exceptions = set()
             bad_keys = set()
-            for key in keys:
-                if key not in self.futures or self.futures[key].status in failed:
+            for future in future_set:
+                key = future.key
+                if key not in self.futures or future.status in failed:
```

Comment:
As opposed to `futures_of` and `wait`, we only raise a single exception here. We should probably align this.

Reply:
We should do this in a follow-up PR, this mixes cancellations with actual errors.